cmake/examples.cmake

```diff
@@ -0,0 +1,15 @@
+# Setup so that examples/CMakeLists.txt will successfully find the local build
+
+get_filename_component(examples_dir "../examples" ABSOLUTE)
+
+# Add examples as an external project.
+# sub_directory cannot be used because the find_package(protobuf)
+# call would cause failures with redefined targets.
+include(ExternalProject)
+ExternalProject_Add(examples
+  DEPENDS libprotobuf protoc
+  PREFIX examples
+  SOURCE_DIR "${examples_dir}"
+  INSTALL_COMMAND "" #Skip
+  CMAKE_CACHE_ARGS "-Dprotobuf_DIR:PATH=${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_CMAKEDIR}"
```

Comment:
CMP0024 refers specifically to using the include command on the output of export(...). It disallows it because in cmake 3+, the file generated by the export command is not reliably available at configuration time. Using ExternalProject_Add avoids this problem because it runs at build-time. I tested this in CMake 2.8.12, it works perfectly.

This is the equivalent to setting up a separate build directory and running cmake separately with protobuf/examples as the source directory.